packages/vscode-extension/src/debug/depsChecker/doctorLogger.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import commonlibLogger, { VsCodeLogProvider } from "../../commonlib/log";
+import { OutputChannel } from "vscode";
+import { DepsLogger } from "@microsoft/teamsfx-core";
+
+export class DoctorLogger implements DepsLogger {
```

Comment:
The log logic seems a bit duplicated. Two loggers are added, `vsc/DoctorLogger` and `core/EmptyLogger`, and the goal is to only output necessary logs. So could the two loggers be merged into one place, since there's already a `doctor` flag?
E.g., 
- Remove `vsc/DoctorLogger` and let `DepsManager` to control which logs to output via `doctor` flag.
- Remove `core/EmptyLooger` and let each checker support `undefined` logger.

Reply:
Well, control log in each checker is fine. But have to handle `cpUtils / test` for `undefined` logger.
My consideration is that current F5 will be replaced with `new F5(with doctor)` and at that point we can just delete `core/EmptyLogger` and `core/VSCodeLogger`(maybe) without changing logic in each checker.

Further, if the output of depManager(and it's checkers) in VSCode and CLI and VS is same, we can move doctorLogger to DepsManager as you said. but so far, I am not sure about that.

Reply:
I see, so `EmptyLogger` sounds good to reduce maintenance effort on each checker.

But `DoctorLogger` still looks weird. I noticed that the only usage of `DoctorLogger` is to append message, so why not directly use `vsc/depsChecker/VSCodeLogger`? My concern is, **since `DoctorLogger` does not implement log-level methods, it brings extra maintenance effort on `core/DepsManager`** - calling `this.logger.info/warning/error` has no output.

New F5 does use doctor, but for backward compatibility, we'll still support original F5 and keep `vsc/depsChecker/VSCodeLogger`, so I feel like no need to create new `DoctorLogger` and could just leverage `VSCodeLogger`.

Reply:
I agree. and will remove DoctorLogger.

Reply:
Removed.